### PR TITLE
muPdf implementation for AA SVGs

### PR DESCRIPTION
**Pull request by codecounselor**
While we likely won't use inkscape I want to leave that code so that it gets into master.  It can be removed at a later time if muPdf ends up being the better option.

### CODE REVIEW

**Comment by mattsgarlata** [comment]
I'm sorry, but I don't understand why we would want the inkscape code if we're not going to use it.  Can we just wait for a couple days for the final solution to reveal itself before merging things in?

File: lib/powerpoint.js [hunk 1]
```diff
@@ -18,13 +19,19 @@ var debugLogger = debug('pdfppt:debug')
...
+/** allows the caller to provide a path to the MuPdf Executable */
+const MUPDF_PATH = process.env.PDF_PPT_MUPDF_PATH
 /** allows the caller to provide a path to the GhostScript Executable */
 const GS_PATH = process.env.PDF_PPT_GSPATH
 /** allows the caller to provide a path to the ImageMagick Executable */
 const IM_PATH = process.env.PDF_PPT_IMPATH
 
-const gsExecutable = os.platform() === 'win32' ? GS_PATH || 'gswin32c.exe' : 'gs'
-const imageMagickConvert = os.platform() === 'win32' ? IM_PATH || 'convert.exe' : 'convert'
+let gsExecutable = GS_PATH || 'gs'
```
Comment (mattsgarlata) [inline lib/powerpoint.js]: Should this be in _setExecutable so all the logic is in one spot?

Comment (codecounselor) [reply lib/powerpoint.js]: yeah good idea

File: lib/powerpoint.js [hunk 2]
```diff
@@ -135,22 +294,58 @@ class Powerpoint extends EventEmitter2 {
...
-      if (!_.isEmpty(gsErr)) {
-        this.emit('err.png.all', {error: gsErr, time: this.elapsed(start)})
+    return this.processImgConversionTasks(tasks, gsErr, start)
+  }
+
+  /**
+   * @param {Array} tasks list of promises
+   * @param {Array} errors
+   * @param {number} startedAt timestamp of when conversion started
+   * @returns {Promise.<string[]>} sorted images ready for pptx slides
```
Comment (mattsgarlata) [inline lib/powerpoint.js]: Not a big deal, but I thought the way to document this was with a capital N for Number and capital S for String

File: lib/powerpoint.js [hunk 3]
```diff
@@ -94,10 +125,141 @@ class Powerpoint extends EventEmitter2 {
...
+   * @param {array} commands inkscape export commands
+   * @returns {Promise}
+   * @private
+   */
+  _spawnInkscapeShell (commands) {
+    return new Promise((resolve, reject) => {
+      const inkProc = spawn(inkScapeExecutable, ['--shell'])
+      inkProc.stdout.on('data', d => {
+        const msg = d.toString()
+        if (_.startsWith(msg, 'Bitmap saved')) {
```
[the remark this answers is not in the document]
Comment (codecounselor) [reply lib/powerpoint.js]: pretty sure it was a buffer, but this is getting deleted in a few mins :)

File: lib/powerpoint.js [hunk 4]
```diff
@@ -228,13 +423,14 @@ class Powerpoint extends EventEmitter2 {
 
   /**
    * @param sortedImages sorted by file and page/image number
-   * @returns {array} of files from sortedImages that are the last image for each file
+   * @returns {array} of files from sortedImages that are the last image for
```
Comment (mattsgarlata) [inline lib/powerpoint.js]: Same thing: I thought it was Array with a capital A

Comment (codecounselor) [reply lib/powerpoint.js]: I can do it that way, I think JSDoc figures it out either way

**Pull request merged**